### ISSUE TITLE
go get install fails

### ISSUE THREAD

**Issue by qhua948** [openshift-online/ocm-cli#235]
Following README.md running
`go get -u github.com/openshift-online/ocm-cli/cmd/ocm` fails with the following error
```
$ go get -u github.com/openshift-online/ocm-cli/cmd/ocm
go: downloading github.com/openshift-online/ocm-cli v0.1.48
go: downloading github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
go: downloading github.com/spf13/pflag v1.0.5
go: downloading github.com/spf13/cobra v1.0.0
go: downloading github.com/openshift-online/ocm-sdk-go v0.1.157
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading gitlab.com/c0b/go-ordered-json v0.0.0-20171130231205-49bbdab258c2
go: downloading github.com/AlecAivazis/survey/v2 v2.2.7
go: downloading github.com/mattn/go-isatty v0.0.12
go: downloading k8s.io/apimachinery v0.18.5
go: downloading github.com/pkg/browser v0.0.0-20180916011732-0a3d74bf9ce4
go: downloading github.com/hashicorp/go-version v1.2.1
go: downloading golang.org/x/sys v0.0.0-20210119212857-b64e53b001e4
go: downloading github.com/hashicorp/go-version v1.3.0
go: downloading github.com/spf13/cobra v1.1.3
go: downloading github.com/inconshreveable/mousetrap v1.0.0
go: downloading github.com/pkg/browser v0.0.0-20210115035449-ce105d075bb4
go: downloading golang.org/x/sys v0.0.0-20210412220455-f1c623a9e750
go: downloading gitlab.com/c0b/go-ordered-json v0.0.0-20201030195603-febf46534d5a
go: downloading github.com/AlecAivazis/survey/v2 v2.2.9
go: downloading github.com/AlecAivazis/survey v1.8.8
go: downloading golang.org/x/crypto v0.0.0-20201217014255-9d1352758620
go: downloading github.com/kballard/go-shellquote v0.0.0-20180428030007-95032a82bc51
go: downloading k8s.io/apimachinery v0.21.0
go: downloading github.com/openshift-online/ocm-sdk-go v0.1.168
go: downloading golang.org/x/crypto v0.0.0-20210322153248-0c34fe9e7dc2
go: downloading github.com/grokify/html-strip-tags-go v0.0.0-20200322061010-ea0c1cf2f119
go: downloading github.com/cenkalti/backoff v2.2.1+incompatible
go: downloading github.com/prometheus/client_golang v1.7.1
go: downloading github.com/cenkalti/backoff/v4 v4.0.0
go: downloading github.com/cenkalti/backoff/v4 v4.1.0
go: downloading golang.org/x/term v0.0.0-20201210144234-2321bbc49cbf
go: downloading github.com/prometheus/client_golang v1.10.0
go: downloading github.com/grokify/html-strip-tags-go v0.0.1
go: downloading github.com/beorn7/perks v1.0.1
go: downloading github.com/cespare/xxhash v1.1.0
go: downloading github.com/golang/protobuf v1.4.2
go: downloading github.com/cespare/xxhash/v2 v2.1.1
go: downloading github.com/prometheus/client_model v0.2.0
go: downloading github.com/prometheus/common v0.11.1
go: downloading github.com/prometheus/procfs v0.1.3
go: downloading golang.org/x/term v0.0.0-20210406210042-72f3dc4e9b72
go: downloading github.com/json-iterator/go v1.1.10
go: downloading github.com/mgutz/ansi v0.0.0-20200706080929-d51e80ef957d
go: downloading github.com/golang/protobuf v1.5.2
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/modern-go/reflect2 v1.0.1
go: downloading google.golang.org/protobuf v1.25.0
go: downloading github.com/prometheus/procfs v0.6.0
go: downloading github.com/prometheus/common v0.20.0
go: downloading github.com/mattn/go-colorable v0.1.8
go: downloading golang.org/x/text v0.3.4
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: downloading google.golang.org/protobuf v1.26.0
go: downloading gopkg.in/yaml.v3 v3.0.0-20200615113413-eeeca48fe776
go: downloading golang.org/x/text v0.3.6
go: downloading gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
go: downloading golang.org/x/net v0.0.0-20210226172049-e18ecbb05110
go: downloading golang.org/x/net v0.0.0-20210410081132-afb366fc7cd1
go: downloading github.com/ghodss/yaml v1.0.0
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading github.com/dgrijalva/jwt-go v1.0.2
# github.com/openshift-online/ocm-cli/pkg/config
../../../go/pkg/mod/github.com/openshift-online/ocm-cli@v0.1.48/pkg/config/config.go:156:24: undefined: sdk.GetTokenExpiry
../../../go/pkg/mod/github.com/openshift-online/ocm-cli@v0.1.48/pkg/config/config.go:173:24: undefined: sdk.GetTokenExpiry
```

**Comment by jhernand** [comment]
That `go get -u ...` command is intended to install the tool without cloning it. But looking at the output of your command I'd say that you cloned the project, and then you run the `go get -u ...` command in the directory of the project. Or maybe you are in the directory of some other project where you have a `go.mod` file. The presence of that `go.mod` file changes the behavior of the `-u` option: it will try to update the versions of the dependencies. In this case that results in a version of the OCM SDK that is incompatible with the version of the CLI. That specific incompatibility was fixed in #233.

If you just want to use the `ocm` tool, then I'd suggest that you get one of the pre-built binary releases from the [releases page](https://github.com/openshift-online/ocm-cli/releases).

If you can't or don't want to use one of the release binaries then you can use that `go get -u ...` command, but do it in a directory where you don't have a `go.mod` file.

Please check and let us know if that solves your problem.

**Comment by qhua948** [comment]
> That `go get -u ...` command is intended to install the tool without cloning it. But looking at the output of your command I'd say that you cloned the project, and then you run the `go get -u ...` command in the directory of the project. Or maybe you are in the directory of some other project where you have a `go.mod` file. The presence of that `go.mod` file changes the behavior of the `-u` option: it will try to update the versions of the dependencies. In this case that results in a version of the OCM SDK that is incompatible with the version of the CLI. That specific incompatibility was fixed in #233.
> 
> If you just want to use the `ocm` tool, then I'd suggest that you get one of the pre-built binary releases from the [releases page](https://github.com/openshift-online/ocm-cli/releases).
> 
> If you can't or don't want to use one of the release binaries then you can use that `go get -u ...` command, but do it in a directory where you don't have a `go.mod` file.
> 
> Please check and let us know if that solves your problem.

Thanks,
I installed it with the binary instead. 

But just to confirm it, I reran the command, but seems to returning the same error
```
$ ls -lah
drwxr-xr-x edward edward 4.0 KB Wed Apr 14 10:30:14 2021  .
drwx------ edward edward 4.0 KB Wed Apr 14 10:30:57 2021  ..
$  go get -u github.com/openshift-online/ocm-cli/cmd/ocm
# github.com/openshift-online/ocm-cli/pkg/config
../go/pkg/mod/github.com/openshift-online/ocm-cli@v0.1.48/pkg/config/config.go:156:24: undefined: sdk.GetTokenExpiry
../go/pkg/mod/github.com/openshift-online/ocm-cli@v0.1.48/pkg/config/config.go:173:24: undefined: sdk.GetTokenExpiry

```

**Comment by jhernand** [comment]
I assume that empty directory is inside another directory that contains a `go.mod` file. Go will look recursively for `go.mod` files in all the parent directories.

As we are already discouraging use of `go get ...` in the README I am closing this.
